### PR TITLE
[DOC-539] [2024.2] [EA] Added colocated tables with tablespaces

### PR DESCRIPTION
**Pull request by aishwarya24**
@netlify /v2024.2/explore/colocation/#colocated-tables-with-tablespaces

Note to self - Move the backup section to back up and restore after confirmation.

### CODE REVIEW

File: docs/content/v2024.2/explore/colocation.md [hunk 1]
```diff
@@ -237,3 +237,132 @@ To set up xCluster for colocated tables, do the following:
...
+```sql
+ALTER TABLE <table_name> SET TABLESPACE <new_tablespace>;
+
+```
+
+### Back up and restore colocated database using ysql_dump
+
+You can back up and restore a database with colocated tables and tablespaces in two ways:
+
+- With `--use_tablespaces` option in [ysql_dump](../../admin/ysql-dump/). Using this option during backup and restore includes tablespace information. The restored database requires the target universe to contain the necessary nodes for tablespace creation.
```
Comment (ddhodge) [inline docs/content/v2024.2/explore/colocation.md:354]: We should add --use_tablespaces to ysql_dump docs at some point

**Review by ddhodge — APPROVED** [review]

**Comment by ddhodge** [comment]
Some variation of "With the introduction of [tablespaces](../going-beyond-sql/tablespaces/) support for colocated tables {{<tags/feature/ea>}}, you can assign tablespaces to their colocated tables, bridging the gap between regional data distribution and performance demands. This development eliminates the need for trade-offs, allowing organizations to adhere to compliance requirements while benefiting from the efficiency of colocated tables and relations. By ensuring all colocated tables in a tablespace reside in the same tablet, this feature optimizes both data management and operational efficiency." would be suitable for RN highlights

**Review by utkarsh-um-yb — CHANGES_REQUESTED** [review]
I have made some additions to the original doc provided to you https://docs.google.com/document/d/1-iCCndnkLbaiftqMlp121i0wsK6ZndIruCO0H3P0PoU/edit, Have highlighted the changes Can you please accommodate  this changes too, thanks!

Also please add @suranjan  to this review as well.

File: docs/content/v2024.2/explore/colocation.md [hunk 2]
```diff
@@ -237,3 +237,132 @@ To set up xCluster for colocated tables, do the following:
...
+
+### Create a colocated table in a tablespace
+
+In a [colocated database](#databases), [tables](#tables) are created with colocation by default. To create a colocated table in a tablespace, use the following command:
+
+```sql
+CREATE TABLE <table_name> TABLESPACE <tablespace_name>;
+```
+
+Use the same syntax to create indexes and materialized views in a tablespace as follows:
```
Comment (utkarsh-um-yb) [inline docs/content/v2024.2/explore/colocation.md:255]:
```suggestion
Use the same syntax to create colocated indexes and materialized views in a tablespace as follows:
```

File: docs/content/v2024.2/explore/colocation.md [hunk 3]
```diff
@@ -237,3 +237,132 @@ To set up xCluster for colocated tables, do the following:
...
+
+You should see output similar to the following:
+
+```output
+  oid  |     grpname      | grpowner | grptablespace | grpacl | grpoptions
+-------+------------------+----------+---------------+--------+------------
+ 16387 | default          |       10 |             0 |        |
+ 16391 | colocation_16384 |       10 |         16384 |        |
+(2 rows)
+```
```
Comment (utkarsh-um-yb) [inline docs/content/v2024.2/explore/colocation.md:303]: Please remove the oid column. The oid column shows up only in pg15, in 2024.2 it won't be there. Sorry this was a miss from my side, in the original doc.

File: docs/content/v2024.2/explore/colocation.md [hunk 4]
```diff
@@ -237,3 +237,132 @@ To set up xCluster for colocated tables, do the following:
...
+
+The `oid` and `grpname` columns represent the tablegroup's OID and name, respectively. The `grptablespace` column shows the OID of the associated tablespace.
```
Comment (utkarsh-um-yb) [inline docs/content/v2024.2/explore/colocation.md:305]: remove the `oid` description from here also.

File: docs/content/v2024.2/explore/colocation.md [hunk 5]
```diff
@@ -237,3 +237,288 @@ To set up xCluster for colocated tables, do the following:
...
+       colocation_16386 | postgres    | t2_3       | table | yugabyte
+       colocation_16386 | postgres    | t1_3       | table | yugabyte
+       default          | postgres    | t2_default | table | yugabyte
+       default          | postgres    | t1_default | table | yugabyte
+       default          | postgres    | t2         | table | yugabyte
+       default          | postgres    | t1 	       | table | yugabyte
+      (10 rows)
+      ```
+
+      Same `Group Name` indicates the colocated entities, and each tablegroup belongs to a different tablespace, as evident from the `grptablespace` column from the following table.
```
Comment (ddhodge) [inline docs/content/v2024.2/explore/colocation.md:471]:
```suggestion
    The Group Name column shows which entities are colocated.
      
     Each tablegroup belongs to a different tablespace, as shown in the grptablespace column in the following table.
```

File: docs/content/v2024.2/explore/colocation.md [hunk 6]
```diff
@@ -237,3 +237,288 @@ To set up xCluster for colocated tables, do the following:
...
+           grpname      | grpowner | grptablespace | grpacl | grpoptions
+      ------------------+----------+---------------+--------+------------
+       default          |       10 |             0 |        |
+       colocation_16384 |       10 |         16384 |        |
+       colocation_16385 |       10 |         16385 |        |
+       colocation_16386 |       10 |         16386 |        |
+      (4 rows)
+      ```
+
+      The same information after backup or restore without the `--use_tablespaces` option would look like the following:
```
Comment (ddhodge) [inline docs/content/v2024.2/explore/colocation.md:487]:
```suggestion
      The same information after backup or restore without the `--use_tablespaces` option looks like the following:
```

File: docs/content/v2024.2/explore/colocation.md [hunk 7]
```diff
@@ -237,3 +237,288 @@ To set up xCluster for colocated tables, do the following:
...
+       colocation_restore_16405 | postgres    | t2_3       | table | yugabyte
+       colocation_restore_16405 | postgres    | t1_3       | table | yugabyte
+       default                  | postgres    | t2_default | table | yugabyte
+       default                  | postgres    | t1_default | table | yugabyte
+       default                  | postgres    | t2         | table | yugabyte
+       default                  | postgres    | t1         | table | yugabyte
+      (10 rows)
+      ```
+
+      As evident from the above table, the colocation property is still maintained after the backup or restore. But all these tables now reside in the same tablespace which is the default one, as per the following table:
```
Comment (ddhodge) [inline docs/content/v2024.2/explore/colocation.md:510]:
```suggestion
    The colocation property is still maintained after the backup or restore. But all the tables now reside in the same tablespace (the default one):
```

File: docs/content/v2024.2/explore/colocation.md (hunk 4 above)
Comment (utkarsh-um-yb) [reply docs/content/v2024.2/explore/colocation.md]: yes we can include that

**Pull request merged**